src/archive.rs

```diff
@@ -197,15 +197,17 @@ impl OpenArchive {
            destination: Option<String>,
            operation: Operation)
            -> UnrarResult<Self> {
-        let mut data = native::OpenArchiveData::new(cstr!(filename).as_ptr() as *const _,
+        let filename = cstr!(filename);
+        let mut data = native::OpenArchiveData::new(filename.as_ptr() as *const _,
                                                     mode as u32);
         let handle = unsafe { native::RAROpenArchive(&mut data as *mut _) };
         let result = Code::from(data.open_result).unwrap();
         if handle.is_null() {
             Err(UnrarError::from(result, When::Open))
         } else {
             if let Some(pw) = password {
-                unsafe { native::RARSetPassword(handle, cstr!(pw).as_ptr() as *const _) }
+                let pw = cstr!(pw);
+                unsafe { native::RARSetPassword(handle, pw.as_ptr() as *const _) }
```

Comment:
I believe this would still be a dangling pointer as the pointer will still point to the memory after it's freed, right?

Reply:
Well, it shouldn't be because `RARSetPassword` seems to copy the buffer (actually it converts it to `wchar`), so it's not storing the pointer beyond this call.